modules/core/src/test/java/org/apache/ignite/session/GridSessionSetTaskAttributeSelfTest.java

```diff
@@ -101,6 +130,41 @@ public void testMultiThreaded() throws Exception {
             fail();
     }
 
+    /**
+     * Check parameters propagation from session to events
+     * @throws Exception
+     */
+    @Test
+    public void testAttributesToEventsPropagation() throws Exception {
+        final int intParam = 1;
+        final String textParam = "text";
+
+        IgniteEx n = startGrid(0);
+
+        n.cluster().state(ACTIVE);
+
+        IgnitePredicate<TaskEvent> lsnr = evt -> {
+            System.out.println("Received task event [evt=" + evt.name() + ", taskName=" + evt.taskName() + ", taskAttributes=" + evt.attributes() + ']');
+
+            if (evt.type() != EVT_TASK_STARTED && evt.attributes().isEmpty()) {
```

Comment:
I think here you need to do `assertEquals(EVT_TASK_STARTED, evt.type())` also `assertFalse(evt.attributes().isEmpty())` and not a condition.

I don’t quite understand how it is checked that the listener has worked accurately and correctly, maybe we can wrap the result in the future and wait for it?

Reply:
I corrected it myself.